src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -202,84 +202,92 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
+          loadConfig.setAutodetect(true);
+        } else {
+          LOG.info("Using provided import schema '{}'.", schema.toString());
+        }
 
-      JobConfiguration config = new JobConfiguration();
-      config.setLoad(loadConfig);
+        JobConfiguration config = new JobConfiguration();
+        config.setLoad(loadConfig);
 
-      // Get the dataset to determine the location
-      Dataset dataset =
-        bigQueryHelper.getRawBigquery().datasets().get(tableRef.getProjectId(), tableRef.getDatasetId()).execute();
+        // Get the dataset to determine the location
+        Dataset dataset =
+          bigQueryHelper.getRawBigquery().datasets().get(tableRef.getProjectId(), tableRef.getDatasetId()).execute();
 
-      JobReference jobReference =
-        bigQueryHelper.createJobReference(projectId, "direct-bigqueryhelper-import", dataset.getLocation());
-      Job job = new Job();
-      job.setConfiguration(config);
-      job.setJobReference(jobReference);
+        JobReference jobReference =
+          bigQueryHelper.createJobReference(projectId, "direct-bigqueryhelper-import", dataset.getLocation());
+        Job job = new Job();
+        job.setConfiguration(config);
+        job.setJobReference(jobReference);
 
-      // Insert and run job.
-      bigQueryHelper.insertJobOrFetchDuplicate(projectId, job);
+        // Insert and run job.
+        bigQueryHelper.insertJobOrFetchDuplicate(projectId, job);
 
-      // Poll until job is complete.
-      waitForJobCompletion(bigQueryHelper.getRawBigquery(), projectId, jobReference);
+        // Poll until job is complete.
+        waitForJobCompletion(bigQueryHelper.getRawBigquery(), projectId, jobReference);
 
-      if (temporaryTableReference != null && bigQueryHelper.tableExists(temporaryTableReference)) {
-        long expirationMillis = System.currentTimeMillis() + TimeUnit.DAYS.toMillis(1);
-        Table table = bigQueryHelper.getTable(temporaryTableReference).setExpirationTime(expirationMillis);
-        bigQueryHelper.getRawBigquery().tables().update(temporaryTableReference.getProjectId(),
-                                                        temporaryTableReference.getDatasetId(),
-                                                        temporaryTableReference.getTableId(), table).execute();
+        if (temporaryTableReference != null && bigQueryHelper.tableExists(temporaryTableReference)) {
+          long expirationMillis = System.currentTimeMillis() + TimeUnit.DAYS.toMillis(1);
+          Table table = bigQueryHelper.getTable(temporaryTableReference).setExpirationTime(expirationMillis);
+          bigQueryHelper.getRawBigquery().tables().update(temporaryTableReference.getProjectId(),
+                                                          temporaryTableReference.getDatasetId(),
+                                                          temporaryTableReference.getTableId(), table).execute();
+        }
+      } else if (!bigQueryHelper.tableExists(tableRef)) {
```

Comment:
When does this happen? Can you add some comments?

Reply:
This block was added for case when gcsPaths list is empty and destination table not exist. In this case empty destination table will be created.